components/molecules/Pagination/pagination.tsx

```diff
@@ -46,22 +52,20 @@ const Pagination = ({
   // These numbers change as you navigate the paging.
   // It might start off as 1,2,3,4,5 but as you move forward it becomes
   // 2,3,4,5,6 and so on.
-
-  const middlePages = Array.from({ length: Math.min(pages.length, pageSize) }, (_, index) =>
-    page <= Math.floor(pageSize / 2)
+  const middlePages = Array.from({ length: Math.min(totalPage, numberOfPagesToDisplay) }, (_, index) =>
```

Comment:
This fixes up some issues with the paging component when we display a list of pages.